buildSrc/src/main/groovy/airbyte-connector-docker-convention.gradle

```diff
@@ -28,10 +28,31 @@ class AirbyteConnectorDockerConventionPlugin implements Plugin<Project> {
             }
 
             /* ------------------------------------------------------------
-             * 1. imageName default = project.name (if not already set)
+             * 1. imageName default = dockerRepository from metadata.yaml (if not already set)
+             *    Falls back to project.name if dockerRepository is not in metadata.yaml
              * ------------------------------------------------------------ */
             if (!dockerExt.imageName.present) {
-                dockerExt.imageName.convention(project.name)
+                def metadataFile = project.layout.projectDirectory.file('metadata.yaml').asFile
+                if (metadataFile.exists()) {
+                    // Read dockerRepository from metadata.yaml
+                    def dockerRepo = null
+                    metadataFile.eachLine { line ->
+                        def matcher = (line =~ /^\s*dockerRepository:\s*(.+)$/)
```

Comment:
should we use a yaml parser library instead? I think we already have `snakeyaml` as a dependency